src/main/java/seedu/address/model/entry/Entry.java

```diff
@@ -0,0 +1,160 @@
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents an Entry in the entry book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Entry {
+
+    // Identity fields
+    private final Link link;
+
+    // Data fields
+    private final Title title;
+    private final Comment comment;
+    // private final Description description;
+    // private final SiteName siteName;
+
+    // private final Set<Feed> feeds = new HashSet<>();
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Entry (Link link,
+                  Title title,
+                  Comment comment,
+                  // Description description,
+                  // SiteName siteName,
+                  // Set<Feed> feeds,
+                  Set<Tag> tags) {
+        requireAllNonNull(link, title, tags);
+        this.link = link;
+        this.title = title;
+        this.comment = comment;
+        // this.description = description;
+        // this.siteName = siteName;
+        // this.feeds = feeds;
+        this.tags.addAll(tags);
+    }
+
+    public Link getLink () {
+        return link;
+    }
+
+    public Title getTitle() {
+        return title;
+    }
+
+    public Comment getComment() {
+        return comment;
+    }
+
+    /*
+    public Description getDescription() {
+        return description;
+    }
+
+    public SiteName getSiteName() {
+        return siteName;
+    }
+    */
+
+    /**
+     * Returns an immutable feed set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    /*
+     public Set<Feed> getFeed() {
+        return Collections.unmodifiableSet(feeds);
+     }
+     */
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both entries have the same link.
+     * This defines a weaker notion of equality between two entries.
+     */
+    public boolean isSameEntry(Entry otherEntry) {
+        if (otherEntry == this) {
+            return true;
+        }
+
+        return otherEntry != null
+                && otherEntry.getLink().equals(getLink());
+    }
+
+    /**
+     * Returns true if both entries have the same identity and data fields.
+     * This defines a stronger notion of equality between two entries.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Entry)) {
+            return false;
+        }
+
+        Entry otherEntry = (Entry) other;
+        return otherEntry.getLink().equals(getLink())
```

Comment:
I think equality check should be on the `Link` field only, so that users cannot add 2 entries with the same URL link.

Reply:
Yup that check is implemented in isSameEntry(), which will be used for checking duplicate entries.

Reply:
Ah yes, I got confused between `isSameEntry()` and `equals()`. Looks good for now.